miniapps/tools/plor-transfer.cpp

```diff
@@ -0,0 +1,375 @@
+   HypreParMatrix* M_lor_tdof = M_lor.ParallelAssemble();
+
+   // HO projections
+   direction = "HO -> LOR @ HO";
+   FunctionCoefficient RHO(RHO_exact);
+   rho.ProjectCoefficient(RHO);
+   double ho_mass = compute_mass(&fespace, -1.0, HO_dc, "HO       ");
+   if (vis) { visualize(HO_dc, "HO", Wx, Wy); Wx += offx; }
+
+   GridTransfer *gt;
+   if (use_pointwise_transfer)
+   {
+      gt = new InterpolationGridTransfer(fespace, fespace_lor);
+   }
+   else
+   {
+      gt = new L2ProjectionGridTransfer(fespace, fespace_lor);
+   }
+   const Operator &R = gt->ForwardOperator();
+
+   // HO->LOR restriction
+   direction = "HO -> LOR @ LOR";
+   R.Mult(rho, rho_lor);
+   compute_mass(&fespace_lor, ho_mass, LOR_dc, "R(HO)    ");
+   if (vis) { visualize(LOR_dc, "R(HO)", Wx, Wy); Wx += offx; }
+   auto global_max = [](const Vector& v)
+   {
+      double max = v.Normlinf();
+      MPI_Allreduce(MPI_IN_PLACE, &max, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+      return max;
+   };
+
+   if (gt->SupportsBackwardsOperator())
+   {
+      const Operator &P = gt->BackwardOperator();
+      // LOR->HO prolongation
+      direction = "HO -> LOR @ HO";
+      ParGridFunction rho_prev = rho;
+      P.Mult(rho_lor, rho);
+      compute_mass(&fespace, ho_mass, HO_dc, "P(R(HO)) ");
+      if (vis) { visualize(HO_dc, "P(R(HO))", Wx, Wy); Wx = 0; Wy += offy; }
+
+      rho_prev -= rho;
+      Vector rho_prev_true(fespace.GetTrueVSize());
+      rho_prev.GetTrueDofs(rho_prev_true);
+      double l_inf = global_max(rho_prev_true);
+      if (Mpi::Root())
+      {
+         cout.precision(12);
+         cout << "|HO - P(R(HO))|_∞   = " << l_inf << endl;
+      }
+   }
+
+   // HO* to LOR* dual fields
+   ParGridFunction ones(&fespace), ones_lor(&fespace_lor);
+   ones = 1.0;
+   ones_lor = 1.0;
+   ParLinearForm M_rho(&fespace), M_rho_lor(&fespace_lor);
+   auto global_sum = [](const Vector& v)
+   {
+      double sum = v.Sum();
+      MPI_Allreduce(MPI_IN_PLACE, &sum, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+      return sum;
+   };
+   if (!use_pointwise_transfer && gt->SupportsBackwardsOperator())
+   {
+      const Operator &P = gt->BackwardOperator();
+      Vector rho_true(rho.ParFESpace()->GetTrueVSize());
+      rho.GetTrueDofs(rho_true);
+      Vector M_rho_true(M_rho.ParFESpace()->GetTrueVSize());
+      M_ho_tdof->Mult(rho_true, M_rho_true);
+      M_rho.ParFESpace()->GetProlongationMatrix()->Mult(M_rho_true, M_rho);
+      P.MultTranspose(M_rho, M_rho_lor);
+      Vector M_rho_lor_true(M_rho_lor.ParFESpace()->GetTrueVSize());
+      M_rho_lor.ParFESpace()->GetRestrictionOperator()->Mult(M_rho_lor,
+                                                             M_rho_lor_true);
+      double ho_dual_mass = global_sum(M_rho_true);
+      double lor_dual_mass = global_sum(M_rho_lor_true);
+      if (Mpi::Root())
+      {
+         cout << "HO -> LOR dual field: " << fabs(ho_dual_mass - lor_dual_mass) << endl
+              << endl;
+      }
```

Comment:
This will have to be adjusted after the action of `MultTranspose` is changed per the previous comment.